test/models/user_test.rb

```
@@ -0,0 +1,84 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  def test_user_new
+    user = User.new(name: 'jon', email: 'jon@example.com', password: 'password1', password_confirmation: 'password1')
+    assert_predicate user, :valid?
+  end
+
+  def test_email_coerced_to_lowercase
+    user = User.new(name: 'jon', email: 'JON@Example.com', password: 'password1', password_confirmation: 'password1')
+    user.save
```

While it's rare, consider using `save!` so that an exception raises in the event this happens to fail. Similar for `create!` and `update!`. The non-`!` should be used only in cases where you are explicitly utilizing the return value.

---

Totally makes sense to fail fast in such a situation. Thanks for the suggestion! Implemented in 7be79965b5ae9fb510e320c8bdd4e83efa6a9b3d.